elisa/src/org/labkey/elisa/HighThroughputImportHelper.java

```diff
@@ -0,0 +1,304 @@
+            analyteToPlate.put(entry.getKey(), plate);
+        }
+        return analyteToPlate;
+    }
+
+    @Override
+    public Map<String, Double> getStandardConcentrations(String plateName, int analyteNum) throws ExperimentException
+    {
+        return _plateMap.get(plateName).getStdConcentrations(analyteNum);
+    }
+
+    private static class AnalytePlate
+    {
+        private Map<Integer, Map<String, Double>> _stdConcentrations = new HashMap<>();
+        private Map<Integer, double[][]> _dataMap = new HashMap<>();
+        private String _plateName;
+        private PlateTemplate _plateTemplate;
+        // contains the mapping of (well/analyte) to extra row data to merge during data import
+        private Map<String, Map<String, Object>> _extraWellData = new HashMap<>();
+        public static final String CONTROL_ID_COLUMN = "Sample";
+
+        public AnalytePlate(String plateName, PlateTemplate plateTemplate)
+        {
+            _plateName = plateName;
+            _plateTemplate = plateTemplate;
+        }
+
+        public Map<String, Double> getStdConcentrations(int analyteNum)
+        {
+            if (_stdConcentrations.containsKey(analyteNum))
+                return _stdConcentrations.get(analyteNum);
+            else
+                return Collections.emptyMap();
+        }
+
+        public Map<Integer, double[][]> getDataMap()
+        {
+            return _dataMap;
+        }
+
+        public String getPlateName()
+        {
+            return _plateName;
+        }
+
+        public void setStdConcentration(Position position, Integer spot, Double concentration)
+        {
+            for (WellGroupTemplate wellGroup : _plateTemplate.getWellGroups(position))
+            {
+                if (wellGroup.getType() == WellGroup.Type.REPLICATE)
+                {
+                    Map<String, Double> concentrations = _stdConcentrations.computeIfAbsent(spot, s -> new HashMap<>());
+                    concentrations.put(wellGroup.getPositionDescription(), concentration);
+
+                    return;
+                }
+            }
+        }
+
+        public void setRawSignal(Position position, Integer spot, Integer signal)
+        {
+            if (position != null && spot != null && signal != null)
+            {
+                double[][] data = _dataMap.computeIfAbsent(spot, s -> new double[_plateTemplate.getRows()][_plateTemplate.getColumns()]);
+                data[position.getRow()][position.getColumn()-1] = signal.doubleValue();
+            }
+        }
+
+        public void setExtraProperties(Map<String, Object> row, PositionImpl position, Integer spot, List<? extends DomainProperty> resultsDomain)
+        {
+            Map<String, Object> extraProperties = new HashMap<>();
+            // need to adjust the column value to be 0 based to match the template locations
+            position.setColumn(position.getColumn()-1);
```

Comment:
this `position.getColumn()-1` code shows up a few times, is it worth adding a helper to the PositionImpl? something like position.getZeroBasedColumn() that just returns column-1?